code/modules/library/lib_machines.dm

```diff
@@ -396,17 +397,32 @@ datum/borrowbook // Datum used to keep track of who has borrowed what when and f
  * Library Scanner
  */
 /obj/machinery/libraryscanner
-	name = "scanner"
+	name = "upload scanner"
+	desc = "A machine that scans books for upload to the library database."
 	icon = 'icons/obj/library.dmi'
 	icon_state = "bigscanner"
 	var/insert_anim = "bigscanner1"
-	anchored = 1
-	density = 1
+	anchored = TRUE
+	density = TRUE
 	var/obj/item/book/cache		// Last scanned book
 
-/obj/machinery/libraryscanner/attackby(var/obj/O as obj, var/mob/user as mob)
+/obj/machinery/libraryscanner/attackby(obj/item/O, mob/user)
 	if(istype(O, /obj/item/book))
+		if(!anchored)
+			to_chat(user, SPAN_WARNING("\The [src] must be secured to the floor first!"))
+			return	
 		user.drop_from_inventory(O,src)
+	if(O.iswrench())
+		playsound(get_turf(src), O.usesound, 75, TRUE)
```

Comment:
`use_tool` instead?

Reply:
`use_tool` is for processes that have a delay, such as welding, or things that involve a `do_after`. Instant actions are not currently supported by `use_tool` as it returns if no delay is set.